internal/data/payments.go

```diff
@@ -91,3 +92,86 @@ func (m *PaymentModel) AddPayment(ctx context.Context, tx db.Transaction, paymen
 
 	return nil
 }
+
+func (m *PaymentModel) GetPaymentsPaginated(ctx context.Context, address string, beforeID, afterID int64, sort SortOrder, limit int) ([]Payment, bool, bool, error) {
+	if !sort.IsValid() {
+		return nil, false, false, fmt.Errorf("invalid sort value: %s", sort)
+	}
+
+	if beforeID != 0 && afterID != 0 {
+		return nil, false, false, errors.New("at most one cursor may be provided, got afterId and beforeId")
+	}
+
+	const filteredSetCTE = `
+		WITH filtered_set AS (
+			SELECT * FROM ingest_payments WHERE $1 = '' OR $1 IN (from_address, to_address)
+		)
+	`
+
+	var selectQ string
+	if beforeID != 0 && sort == DESC {
+		selectQ = fmt.Sprintf("SELECT * FROM (SELECT * FROM filtered_set WHERE operation_id > %d ORDER BY operation_id ASC LIMIT $2) AS reverse_set ORDER BY operation_id DESC", beforeID)
```

Comment:
This is a security issue, we should never use external values directly in our queries. Here we should use the `?` operator, then we can use the `Rebind` function to translate it to `$n`.

Reply:
See [b3aac4f](https://github.com/stellar/wallet-backend/pull/28/commits/b3aac4f2478549997b8c8e0eed4c5584623cea0f)